### ISSUE TITLE
check_http always eats last -k 'Host: ' header

### ISSUE THREAD

**Issue by bahbarnett** [monitoring-plugins/monitoring-plugins#1598]
I did search the tracker, but didn't see anything that would be a direct hit on this.

FYI, I'm running Debian Stretch's version of monitoring plugins.
# /usr/lib/nagios/plugins/check_http -V
check_http v2.2 (monitoring-plugins 2.2)

What happens, is that -k works perfectly.. unless you are using -k 'Host:'.  It will always eat the *last* instance -k 'Host:'.

Example lines:
---------------------

# /usr/lib/nagios/plugins/check_http -S -I '1.1.1.1' -k 'Host: 1some.hostname'  -k 'Host: 5some.hostname'  -k 'Host: 3some.hostname'  -u '/register/index.php' -s 'thingtofind' --expect=200  -v | head -n 8
SSL initialized
GET /register/index.php HTTP/1.0
User-Agent: check_http/v2.2 (monitoring-plugins 2.2)
Connection: close
Host: 1some.hostname
Host: 5some.hostname

# /usr/lib/nagios/plugins/check_http -S -I '1.1.1.1' -k 'Host: 2some.hostname' -u '/register/index.php' -s 'thingtofind' --expect=200  -v | head -n 8
SSL initialized
GET /register/index.php HTTP/1.0
User-Agent: check_http/v2.2 (monitoring-plugins 2.2)
Connection: close

root@meeseeks:/etc/icinga# /usr/lib/nagios/plugins/check_http -S -I '1.1.1.1' -k 'host: 2some.hostname' -u '/register/index.php' -s 'thingtofind' --expect=200  -v | head -n 8
SSL initialized
GET /register/index.php HTTP/1.0
User-Agent: check_http/v2.2 (monitoring-plugins 2.2)
Connection: close
host: 2some.hostname

I'm guessing a lot of people just use -k 'host:', neglecting the capital H in the line. 

FYI, in the triple -k line above, check_http shows it sending two Host lines.  Apache gets angry, and responds with 'Bad request'.

In other words?  It seems like host: is replacing Host: still, but Host: gets eaten?

**Comment by ChrisSLR** [comment]
It's a bug in my opinion. Using a force_host_header shouldn't depend on host_name as it does in line 970 of check_http.c and should account for multiple usage.

What happens:
In line 962 the code loops over the headers provides by -k and copies every header starting with "Host:" to "force_host_header". The "winning" value of force_host_header is the last Host-Header provided by -k.

        force_host_header = http_opt_headers[i];

Before writing force_host_header to the send-buffer it checks for "host_name" (-H) which isn't set in your case. This means, that force_host_header is not written to the send-buffer at all if no host_name is provided.

  if (host_name) {
    if (force_host_header) {
      xasprintf (&buf, "%s%s\r\n", buf, force_host_header);

Later the code loops over the headers again, writing every header to the buffer, exept the "force_host_header" discovered before:

      if (force_host_header != http_opt_headers[i]) {
        xasprintf (&buf, "%s%s\r\n", buf, http_opt_headers[i]);



If you're looking for a workaround add a "dummy" Header as a "last" header that is OK to be dropped as the unusable "force_host_header".

You can also change "Host:" to "HOST:" or "host:" to by-pass the host-header recognition. In this case the code will insert it's own Host-Header based on the IP (e.g. "Host: 1.1.1.1") which might trigger an undesired behaviour with the webserver.
